syft/pkg/cataloger/binary/classifier.go

```diff
@@ -79,6 +80,11 @@ func fileNameTemplateVersionMatcher(fileNamePattern string, contentTemplate stri
 
 		filepathNamedGroupValues := internal.MatchNamedCaptureGroups(pat, location.RealPath)
 
+		// versions like 3.5 should not match any character, but explicit dot
```

Comment:
Would it be possible to add an example to this comment? I'm a little unclear what case this should be covering.

Is it "3.5 should not be captured and instead the things that should be matched are 3\\.5\\.0"?

Reply:
I'm happy to try to make this comment more clear somehow but yes, it's saying when the version is passed to the regex template, `3.5` should be treated as `3\.5`